### ISSUE TITLE
Auth User Information Collecting Assumes Fields Exist

### ISSUE THREAD

**Issue by freshleafmedia** [itsgoingd/clockwork#301]
In the PR #296 it is assumed that the `$user` object returned from $this->app['auth']->user() has `email`, `name` and `id` properties but this isn't necessary the case. For example when using a custom user provider

**Comment by motia** [comment]
I am using doctrine in laravel, and I got the same issue because the fields are private in my entities.

**Comment by itsgoingd** [comment]
Yeah, this is aimed at the most common use-case with default Laravel configuration.

You can always set the user yourself like this:

```php
clock()->getRequest()->setAuthenticatedUser($userName, $id, [
 	'email' => $email,
 	'name'  => $fullName
 ]);
```

@motia oops, we definitively don't want to be crashing, I'll add a couple of isset-checks

**Comment by motia** [comment]
Thanks @itsgoingd. this will do it,

**Comment by freshleafmedia** [comment]
I can confirm this has been fixed in v3.1.1. More specifically 6a8f17bcd177b17ce3718972ea3d2b4c8661be01

**Comment by zarunet** [comment]
This is broken in v4.0.5 (and probably all v4.x versions before it).

The commit fixing it seems to have been made on the `v3` branch and wasn't applied onto `master`.

**Comment by websmurf** [comment]
Can confirm this is broken in the 4.x branch again.

**Comment by zarunet** [comment]
That's weird, that file hasn't been touched since my PR... Are you sure you're on 4.0.6 or above?

**Comment by websmurf** [comment]
Ah I see what I did wrong, i've installed 4.0.2 as it's marked as the latest available version (upgrading from 3.x). Missed the "Show 5 newer tags" as well: 

![image](https://user-images.githubusercontent.com/4027236/64046112-5173e600-cb6b-11e9-94e2-c853bf40fb1d.png)

Its fixed with 4.0.7 indeed :-)

**Comment by itsgoingd** [comment]
My bad, I always forget to fill those in, need a better release process. Btw. I'd suggest `^4.0` version constraint for composer, we don't break stuff in minor versions.
